File: src/main/java/shopping/wish/domain/WishList.java
package shopping.wish.domain;

import jakarta.persistence.*;
import shopping.member.domain.Member;
import shopping.product.domain.Product;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
public class WishList {

    @Id
    @GeneratedValue
    private Long id;

    @OneToOne
    @JoinColumn(name = "member_id")
    private Member member;

    @OneToMany(mappedBy = "wishList", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<WishListItem> items = new ArrayList<>();

    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;

    protected WishList() {

    }

    public static WishList create(Long id) {
        return new WishList();
    }

    public Long getId() {
        return id;
    }

    public void addItem(Product product) {
        WishListItem item = WishListItem.create(this, product);
        items.add(item);
    }
}

File: src/main/kotlin/shopping/wish/domain/WishList.kt
package shopping.wish.domain

import jakarta.persistence.Column
import jakarta.persistence.Embedded
import jakarta.persistence.Entity
import jakarta.persistence.Table
import shopping.global.common.BaseEntity

@Entity
@Table(name = "wish_list")
class WishList(
    memberId: Long
) : BaseEntity() {
    @field:Column(name = "member_id", nullable = false)
    var memberId: Long = memberId
        protected set

    @field:Embedded
    private val wishProducts: WishProducts = WishProducts()

    fun addWishProduct(wishProduct: WishProduct) {
        this.wishProducts.add(wishProduct)
    }

    fun deleteWishProduct(wishProduct: WishProduct) {
        this.wishProducts.delete(wishProduct)
    }

    fun getWishProducts(): Set<WishProduct> = wishProducts.toSet()
}
